### ISSUE TITLE
[REQUEST] Add marker buttons for silent sections of a sonog

### ISSUE THREAD

**Issue by BunnyTub** [Founntain/osuplayer#286]
**Is your feature request related to a problem? Please describe.**
My feature request is NOT a problem, rather a feature.

**Describe the solution you'd like**
There isn't a way to skip silent sections now, so I'm thinking about some form of marker start, end, and remove buttons to mark places in the song that are silent (not part of the song, too long, etc)

**Describe alternatives you've considered**
I have not considered any other.

**Additional context**
I've made a mockup of a design.
![timeline](https://github.com/user-attachments/assets/472b4f6d-7553-469a-bf4f-f5250853d523)

**Comment by Founntain** [comment]
Could you elaborate what the `+` `-` and `x` button should do?

**Comment by BunnyTub** [comment]
The + (add) button should add a point that indicates the start and end of silence.
The - (remove) button should remove any point where the timeline currently stands.
The x (trash/delete all) button should remove all points, and is basically a clear button.

Just a mockup, it can probably be done better a different way.

**Comment by Founntain** [comment]
I see, so those points are placed by the user itself or automatically? And if I press the `+` it adds the a marker to the current position. Can there be more than 2 points? I assume not?

And what I want to ask as well? What do you define as silent sections? Do you mean the sections that could be skipped at the beginning / end of song in the game?

**Comment by BunnyTub** [comment]
I see, so those points are placed by the user itself or automatically?
- The points are placed by the user. (I've thought about automatically placing points, but that would cause audio processing on every song, and I'm not sure if that's a good idea, or how to work with that. Maybe something could be done there, but for now, it should be kept simple)

And if I press the + it adds the a marker to the current position.
- Yes.

Can there be more than 2 points?
- No. (unless you add another button for end of silence, and change the add marker button to start of silence)

What do you define as silent sections?
- I define these sections as complete silence, that isn't apart of the song, usually only in the beginning, and the end of the audio.

Do you mean the sections that could be skipped at the beginning / end of song in the game?
- No.

**Comment by BunnyTub** [comment]
And sorry for the long wait on the reply. I've been busy the past few days.

**Comment by Founntain** [comment]
I see. Thanks for the indepth explaination, this helps a lot. I see what we can do, but don't expect anything soon or maybe at all. I keep you updated here

**Comment by BunnyTub** [comment]
Sounds good
